### PR TITLE
Require fields on veteran rep search

### PR DESCRIPTION
**Pull request by charleystran**
## Description of change
This PR adds the proper error handling for when the rep search endpoint is called without first name and last name
## Testing done
Local testing

## Acceptance Criteria (Definition of Done)
- [ ]  when no first or last name params are in the api call, a 422 is returned with a message

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

resolve https://github.com/department-of-veterans-affairs/vets-contrib/issues/4110

### CODE REVIEW

**Comment by edmangimelli** [comment]
LGTM 🎉

File: modules/veteran/app/controllers/veteran/v0/representatives_controller.rb [hunk 1]
```diff
@@ -17,6 +18,18 @@ def find_rep
...
         end
       end
+
+      private
+
+      def check_required_fields
+        if params[:first_name].blank? || params[:last_name].blank?
+          render json: {
+            errors: [
+              details: 'First name and Last name are required to complete this request', status: 422
```
Comment (edmangimelli) [inline modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: do you both default to `details` for errors? (over title)

Comment (bastosmichael) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: It's technically detail (no s) and it's part of the JSON API spec error response https://jsonapi.org/format/#errors

Comment (edmangimelli) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: we fell into the habit of using `title` for general messages, and `details` when the message included unique info (to the request) (that's how we interpreted jsonapi)

Comment (charleystran) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: @bastosmichael this has been updated

Comment (edmangimelli) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: still looks like `details` to me

Comment (edmangimelli) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: do we never use `title` or `code`?

Comment (charleystran) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: if you go into files_changed it should show as detail:

Comment (edmangimelli) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: weird. that's where i thought i was looking 🤷‍♀

**Comment by charleystran** [comment]
@bastosmichael I updated the error hash to include the extra fields

File: modules/veteran/app/controllers/veteran/v0/representatives_controller.rb [hunk 2]
```diff
@@ -17,6 +18,24 @@ def find_rep
...
+      def check_required_fields
+        errors = []
+        errors << error_hash('first_name') if params[:first_name].blank?
+        errors << error_hash('last_name') if params[:last_name].blank?
+        render json: { errors: errors }, status: 422 if errors.any?
+      end
+
+      def error_hash(parameter)
+        {
+          detail: "#{parameter.humanize} are required to complete this request",
```
Comment (edmangimelli) [inline modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: should `are` be `is`?

**Comment by edmangimelli** [comment]
does this resolve https://github.com/department-of-veterans-affairs/vets-contrib/issues/4110
?

**Comment by bastosmichael** [comment]
yes @edmangimelli this does resolve that ticket, I usually put the Resolves in the description myself but @charleystran sometimes forgets to do so.

File: modules/veteran/app/controllers/veteran/v0/representatives_controller.rb (hunk 2 above)
Comment (bastosmichael) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: Good catch yes it should

Comment (charleystran) [reply modules/veteran/app/controllers/veteran/v0/representatives_controller.rb]: This has been updated

**Pull request merged**